src/modify.js

```diff
@@ -6,25 +6,69 @@ const config = require("./config").config;
 
 const isImportMap = config && config.manifestFormat === "importmap";
 
-function getMapFromManifest(manifest) {
-  return isImportMap ? manifest.imports : manifest.sofe.manifest;
+function getScopeFromManifest(manifest, scope = "imports") {
+  if (!isImportMap) {
+    if (scope !== "imports") {
+      throw new Error("Sofe implementations can only support imports");
+    }
+    return manifest.sofe.manifest;
+  } else {
+    return manifest[scope] || {};
+  }
 }
 
 function getEmptyManifest() {
-  return isImportMap ? { imports: {} } : { sofe: { manifest: {} } };
+  return isImportMap ? { imports: {}, scopes: {} } : { sofe: { manifest: {} } };
 }
+const deepCopy = (json) => JSON.parse(JSON.stringify(json));
 
-exports.getEmptyManifest = getEmptyManifest;
+exports.modifyMultipleServices = function (env, newImports) {
+  return modifyMultiple(env, newImports, "imports");
+};
 
 exports.modifyService = function (env, serviceName, url, remove) {
+  return modifyKeyValue(env, serviceName, url, remove, "imports");
+};
+
+exports.modifyMultipleScopes = function (env, newImports) {
+  return modifyMultiple(env, newImports, "scopes");
+};
+
+exports.modifyScope = function (env, serviceName, url, remove) {
+  return modifyKeyValue(env, serviceName, url, remove, "scopes");
+};
+
+function modifyMultiple(env, values, manifestScope) {
+  return modifyLock(env, (json) => {
+    const imports = getScopeFromManifest(json, manifestScope);
+    imports[manifestScope] = values;
+    return imports;
+  });
+}
+
+function modifyKeyValue(env, key, value, remove, manifestScope) {
```

Comment:
Similar thing here - imports and scopes are different enough that they should have their own implementation.